omero/developers/json-api.txt

```diff
@@ -0,0 +1,1329 @@
+          "WellSamples": [
+            {
+              "PositionX": {
+                "Symbol": "reference frame",
+                "Value": 21864.47,
+                "@type": "TBD#LengthI",
+                "Unit": "REFERENCEFRAME"
+              },
+              "PositionY": {
+                "Symbol": "reference frame",
+                "Value": 36711.98,
+                "@type": "TBD#LengthI",
+                "Unit": "REFERENCEFRAME"
+              },
+              "omero:details": {
+                # omitted for brevity
+              },
+              "Image": {
+                "Name": "plate1.HTD [Well E02 Field #1]",
+                "AcquisitionDate": 1252939626000,
+                "omero:details": {
+                  # omitted for brevity
+                },
+                "url:image": "http://localhost:4080/api/v0/m/images/2942/",
+                "omero:series": 120,
+                "@id": 2942,
+                "@type": "http://www.openmicroscopy.org/Schemas/OME/2016-06#Image",
+                "Description": "Scan Time: Mon Sep 14 11:36:58 2009"
+              },
+              "PlateAcquisition": {
+                "omero:details": {
+                  # omitted for brevity
+                },
+                "MaximumFieldCount": 4,
+                "StartTime": 1252938959000,
+                "EndTime": 1252939813000,
+                "@id": 102,
+                "@type": "http://www.openmicroscopy.org/Schemas/OME/2016-06#PlateAcquisition"
+              },
+              "@id": 203,
+              "@type": "http://www.openmicroscopy.org/Schemas/OME/2016-06#WellSample"
+            }
+          ]
+        }
+      ]
+    }
+
+It is also possible to list all Wells without filtering by Plate, using the top-level
+URL ``/api/v0/m/wells/`` optionally filtering by the ``plate`` query parameter.
+
+
+List Wells by WellSample Index
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To list Wells in a Plate, loading only a *single* WellSample and Image per Well,
+you can filter by WellSample Index. This list of Wells will not include empty
+Wells (Wells that have no WellSamples and Images).
+
+::
+
+    GET   /api/v0/m/plates/{plate_id}/wellsampleindex/{index}/wells/
+
+
+It is also possible to use the Plate Acquisition ID instead of
+Plate ID, when the WellSample (field) at the specified index was acquired
+as part of that Plate Acquisition:
+
+::
+
+    GET   /api/v0/m/plateacquisitions/{plateacquisition_id}/wellsampleindex/{index}/wells/
+
+
+Get a single Well
+^^^^^^^^^^^^^^^^^
+
+When a single Well is loaded, this will include all the WellSamples and Images
+with Pixels loaded.
+
+::
+
+    GET   /api/v0/m/wells/{well_id}/
+
+
+Creating and Saving Objects
+---------------------------
+
+The JSON API currently supports creating and saving of a limited number of
+object types, namely Projects, Datasets and Screens.
+This is due to technical issues with saving unloaded objects, such as an Image
```

Comment:
I will rephrase it so it does not sound that bad
"It is not yet possible to save objects with unloaded objects e.g. ..."